aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPluginTest.kt

```diff
@@ -153,16 +186,19 @@ class RealAWSCognitoAuthPluginTest {
         }
         // WHEN
         plugin.updatePassword("old", "new", onSuccess, onError)
-        Thread.sleep(1_000)
+        latch.await()
```

Comment:
`latch.await()` might infinitely block the test if things went wrong. We might want to fail early if possible. Maybe something like:

[assertTrue { latch.await(5, TimeUnit.SECONDS) }](url)